distributed/scheduler.py

```diff
@@ -164,6 +164,10 @@ def nogil(func):
     Py_ssize_t, parse_bytes(dask.config.get("distributed.scheduler.default-data-size"))
 )
 
+RESOLVE_WORKER_HOSTNAME = dask.config.get(
+    "distributed.scheduler.resolve-worker-hostname"
+)
+
```

Comment:
I suggest to not define this as a global variable but simply read out the value when required. By defining this as a global variable, the value is fixed for the lifetime of the python interpreter and you can no longer modify it. This is very likely the reason why your tests fail

Reply:
distributed.yaml is read right after the interpreter start and before this constant is defined. whatever you define in tests is then ignored which is why `with dask.config.set` has no effect

Reply:
@oshadura does this answer your question above?